.github/workflows/scripts/gh-pages-prepare.sh

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+
+# Set the error handling options
+set -eu
+
+# Define some emoji for the messages
+check_mark="\xE2\x9C\x85"
+cross_mark="\xE2\x9D\x8C"
+warning_mark="\xE2\x9A\xA0"
+info_mark="\xE2\x84\xB9"
+
+# Define some colors for the messages
+green="\033[0;32m"
+red="\033[0;31m"
+yellow="\033[0;33m"
+blue="\033[0;34m"
+reset="\033[0m"
+
+# Define a function to parse the command-line options
+parse_options() {
+    # Use getopt to handle long and short options
+    local options
+    options=$(getopt -o d --long dry-run -- "$@") || exit 1
+    eval set -- "$options"
+
+    # Use a case statement to handle different options
+    local dry_run=false # Initialize the dry run flag to false
```

Comment:
I don't see much usefulness if dry-run give how much it complicates the script. Any changes are versioned anyway.

Reply:
dry-run was to ease the development for me, it wasn't meant for production.